docs/GraFx-Developers/connectors/build-media-connector/index.md

```diff
@@ -0,0 +1,284 @@
+```
+
+You can now execute the command (in the terminal, in the right directory)
+
+To understand what the CLI can do, you can now execute
+This will give you an overview of the available commands
+
+``` csh
+npx connector-cli
+```
+
+
+Initialize: Means building the Boiler Plate. This is the basic code you'll need to get started.
+
+``` csh
+npx connector-cli init
+```
+
+You now have the basic code. In that code, we have defined some dependencies. To download these, execute the install command.
+
+
+``` csh
+npm install
+```
+
+The result should look something like this
+
+```csh
+added 112 packages, and audited 113 packages in 5s
+
+35 packages are looking for funding
+  run `npm fund` for details
+
+found 0 vulnerabilities
+
+```
+
+Now it's time to create your custom code.
+
+This happens in the file "connector.ts". This is a typescript files.
+
+By default, there will be no implementations. Your IDE can do this for you, to build the default interface methods.
+
+You can fix this, by applying the quick fix "Implement interface 'Media Connector'"
+
+![ui](connector01.png)
+
+The code will be added with all the functions that need to be implemented.
+
+![ui](connector01.png)
+
+
+```typescript
+import { Media } from '@chili-publish/studio-connectors';
+import { Dictionary, QueryOptions, ArrayBufferPointer, ConnectorConfigValue } from '@chili-publish/studio-connectors/dist/Connector.Shared';
+
+  export default class MyConnector implements Media.MediaConnector {
+    detail(id: string, context: Dictionary): Promise<Media.MediaDetail> {
+      throw new Error('Method not implemented.');
+    }
+    query(options: QueryOptions, context: Dictionary): Promise<Media.MediaPage> {
+      throw new Error('Method not implemented.');
+    }
+    download(id: string, previewType: Media.DownloadType, intent: Media.DownloadIntent, context: Dictionary): Promise<ArrayBufferPointer> {
+      throw new Error('Method not implemented.');
+    }
+    getConfigurationOptions(): ConnectorConfigValue[] {
+      throw new Error('Method not implemented.');
+    }
+    getCapabilities(): Media.MediaConnectorCapabilities {
+      throw new Error('Method not implemented.');
+    }
+      
+  }
+  
+```
+
+The definition of what needs to be implemented, can be reached by right clicking on the interfact:
+
+![ui](connector02.png)
+
+!!! Important
+	The name of your class does not really matter. As long as it's defined as the default class. Once deployed, CHILI GraFx will look for the default class to use.  
+	
+	In the above example it's called "MyConnector". You can name it "MyAwesomeConnector"
+
+## Building
+
+If your code works (syntax is right, since we did not add any functionality yet), it will build.
+Therefore you need to give the command to build with the cli-tool "npx".
+
+```zsh
+npx connector-cli build
+```
+
+resulting in
+
+```zsh
++0.045s connector-cli v1.0.49
++0.046s Running command: 'build' with options: {"connectorFile":"./connector.ts","options":{}}
```

Comment:
in context of https://github.com/chili-publish/studio-connector-contrib/pull/15, we hide `connector-cli v` and `Running command...` output (as it's debug info). To be able see them, you need explicitly path `-v` (verbose) flag. I.e. `npc connector-cli build -v`. But again, please consider `-v` as something when you need to debug the behaviour. It applies for other commands as well

Reply:
Future update